src/modules/noit_lua_libnoit_binding.c

```diff
@@ -64,11 +68,71 @@ static int
 lua_noit_checks_subscribe(lua_State *L) {
   return lua_noit_checks_adjustsubscribe(L, 1);
 }
+
 static int
 lua_noit_checks_unsubscribe(lua_State *L) {
   return lua_noit_checks_adjustsubscribe(L, -1);
 }
 
+mtev_hook_return_t
+hook_metric_subscribe_all(void *closure, noit_metric_message_t *m, int *w, int wlen) {
+  int *lane = (int*) closure;
+  assert(*lane < wlen);
+  w[*lane] = 1;
+  return MTEV_HOOK_CONTINUE;
+}
+
+static int
+lua_noit_metric_subscribe_all(lua_State *L) {
+  int *lane = malloc(sizeof(int));
+  *lane = noit_metric_director_my_lane();
+  // Q: Can we register the same hook multiple times (with different closure data)?
+  metric_director_want_hook_register("metrics_select", hook_metric_subscribe_all, lane);
+  return 0;
+}
+
+// account_set is a map: account_id => account_interests
+// account_interests is an array of integers, such that account_interests[lane] > 0
+//   if we should deliver messages for the given account_id to the lane.
+static mtev_hash_table *account_set = NULL;
+
+mtev_hook_return_t
+hook_metric_subscribe_account(void *closure, noit_metric_message_t *m, int *w, int wlen) {
+  if(account_set == NULL) {
+    return MTEV_HOOK_ABORT;
+  }
+  int account_id = m->id.account_id;
+  int *account_interests;
+  if(mtev_hash_retrieve(account_set, (const char *)&account_id, sizeof(account_id), (void **)&account_interests)){
+    for(int i=0; i<wlen; i++){
+      w[i] = account_interests[i];
+    }
+  }
+  return MTEV_HOOK_CONTINUE;
+}
+
+static int
+lua_noit_metric_subscribe_account(lua_State *L) {
+  // TODO: Call this during startup. Where to put this?
+  if(account_set == NULL) {
+    libnoit_init();
+  }
+  int account_id = luaL_checkint(L, 1);
+  int *account_interests = NULL;
+  if(!mtev_hash_retrieve(account_set, (const char *)&account_id, sizeof(account_id), (void **)&account_interests)) {
+    int nthreads = eventer_loop_concurrency();
+    account_interests = calloc(nthreads, (sizeof(*account_interests)));
+    int rc = mtev_hash_store(account_set, (const char *)&account_id, sizeof(account_id), (void **)account_interests);
```

Comment:
This address is on stack... it will not work.  You need a copy of this:
```
int *account_id_copy = calloc(1, sizeof(int));
*account_id_copy = account_id;
mtev_hash_store(account_set, account_id_copy, sizeof(int), (void **)account_interests);
```